roles/dual_nic/tasks/main.yml

```diff
@@ -0,0 +1,36 @@
+- name: Gather ip_subnet4
+  ansible.builtin.shell:
+    cmd: set -o pipefail && ip -4 -oneline addr |  egrep -v '(lo|link|wlp) ' | cut -d . -f 4 | cut -d / -f 1
```

Comment:
I think we get these variables for free if we do gather some facts here:
https://docs.ansible.com/ansible/latest/collections/ansible/builtin/setup_module.html

I suspect `all_ipv4_addresses` would have that info.

Reply:
well ansible_host would probably work too.   But you still need the shell to cut it down to just the last octet.

Reply:
also running 
```
ansible localhost  -m ansible.builtin.setup -a 'filter=facter_*' |grep all_ipv4_addresses
[WARNING]: No inventory was parsed, only implicit localhost is available
```
did not match any facter.  and `facter_networking.interfaces` had many interfaces, appearing in alphabetic order.